cmd/tailscale/cli/cli.go

```diff
@@ -125,16 +125,20 @@ func CleanUpArgs(args []string) []string {
 }
 
 // Run runs the CLI. The args do not include the binary name.
-func Run(args []string) error {
+func Run(args []string) (err error) {
 	if len(args) == 1 && (args[0] == "-V" || args[0] == "--version") {
 		args = []string{"version"}
 	}
-	if runtime.GOOS == "linux" && distro.Get() == distro.Gokrazy && len(args) == 0 &&
-		os.Getenv("GOKRAZY_FIRST_START") == "1" {
-		// Exit with 125 otherwise the CLI binary is restarted
-		// forever in a loop by the Gokrazy process supervisor.
-		// See https://gokrazy.org/userguide/process-interface/
-		os.Exit(125)
+	if runtime.GOOS == "linux" && distro.Get() == distro.Gokrazy {
+		defer func() {
+			// Exit with 125 otherwise the CLI binary is restarted
+			// forever in a loop by the Gokrazy process supervisor.
+			// See https://gokrazy.org/userguide/process-interface/
+			if err != nil {
+				log.Println(err)
+			}
+			os.Exit(125)
```

Comment:
Can we only do this when our parent is the gokrazy process supervisor and not the breakglass shell?

If I'm in the breakglass shell, I don't want any special behavior here.

The `GOKRAZY_FIRST_START` condition provided that property before, but now it's gone.

/cc @stapelberg

Reply:
You could just keep the GOKRAZY_FIRST_START check around, I think?

Reply:
Yeah, works for me.

Reply:
Fixed and commit message updated. All good?